python/cog/director/monitor.py

```diff
@@ -0,0 +1,42 @@
+import threading
+import time
+
+from opentelemetry import trace
+
+from .director import Director
+
+UTILIZATION_SPAN_INTERVAL = 15
+
+
+class Monitor:
+    def __init__(self, *, director: Director):
+        self._director = director
+        self._thread = None
+        self._should_exit = threading.Event()
+        self._tracer = trace.get_tracer("cog-director")
+
+    def start(self) -> None:
+        self._thread = threading.Thread(target=self._run)
+        self._thread.start()
+
+    def stop(self) -> None:
+        self._should_exit.set()
+
+    def join(self) -> None:
+        if self._thread is not None:
+            self._thread.join()
+
+    def _run(self) -> None:
+        last_span_at = time.perf_counter() - UTILIZATION_SPAN_INTERVAL
+        while not self._should_exit.is_set():
+            if time.perf_counter() >= last_span_at + UTILIZATION_SPAN_INTERVAL:
+                idle = self._director.idle.is_set()
+                with self._tracer.start_as_current_span(
+                    name="cog.director.idle"
```

Comment:
Would `cog.director.utilization` be a more descriptive name for this?

Reply:
Sure, but that changes things a bit... Probably better though. Was probably a bit too biased by what we do so far